doc/debugging.rst

```diff
@@ -14,33 +14,88 @@ This allows breakpoints and the execution flow to be viewed and controlled right
+There are multiple means of setting and clearing normal and conditional breakpoints for any line in a file opened in the :doc:`editor`.
+
+  * By selecting the respective option from the Debug menu.
+  * Through pressing a configurable keyboard shortcut (:kbd:`F12` for normal, or :kbd:`Shift-F12` for conditional breakpoints by default).
+  * By clicking to the left of the line number in an open file in the Editor and :kbd:`Shift`-clicking for conditional breakpoints.
+  * With the ``breakpoint()`` function in your code.
+  * Interactively, using the ``!b`` command in an ``ipdb`` session.
+
+GIF: setting conditional breakpoint shift-clicking
+
+You can access and edit local and global variables at each breakpoint through the :doc:`variableexplorer`.
+
+SCREENSHOT with console and variable explorer showing local and global variables
+
+
+
+=================
+Advanced features
+=================
 
-* Full GUI control over debugger execution from the :guilabel:`Debug` menu, :guilabel:`Debug toolbar` and via configurable keyboard shortcuts, along with the standard ``ipdb`` console commands.
+You can avoid stepping through other Python packages while debugging by enabling the new :guilabel:`Ignore Python libraries while debugging` option in Spyder's preferences under :menuselection:`IPython Console --> Debugger --> Debug`.
+This will skip all the built-in and third-party Python modules.
 
-.. image:: images/debugging/debugging-condbreakpoint.png
-   :alt: Inset of Spyder's Editor, with a breakpoint set and the condition dialog open
+SCREENSHOT with option in preferences
 
-* Highlighting of the current frame (debugging step) in the :doc:`editor`.
+When debugging, if your code has variables with the name of Pdb commands, your debugging workflow might be interrupted.
+To avoid the mix of Pdb commands and Python statements when using the interactive prompt add an exclamation mark in front of the Pdb commands to separate them from Python expressions.
+
+SCREENSHOT with "b" variable
+
+You can execute a snippet of code before any Pdb command to make sure you have the modules and variables in your context when debugging.
+To set this up, go to :menuselection:`Preferences --> IPython Console --> Debugger --> Run code while debugging` and write there the code that you want to be executed before the Pdb commands.
+
+SCREENSHOT with Run code while debugging in preferences.
+
+
+
+==================
+Matplotlib support
+==================
 
-* The ability to access and edit local and global variables at each breakpoint through the :doc:`variableexplorer`, and run many commands in the :doc:`ipythonconsole`.
+Plotting Matplotlib figures is fully supported while debugging with the :guilabel:`inline` backend and the :guilabel:`interactive` backends which you can use by typing the ``%matplotlib`` magic.
```

Comment:
```suggestion
Generating Matplotlib figures is fully supported while the debugger is active, including all the different graphics backends.
Use the ``%matplotlib`` magic to change to an interactive backend (e.g. ``%matplotlib qt5``) to pan, zoom and adjust your plots in a separate window, or switch back to the default ``inline`` (``%matplotlib inline``) to see them displayed right in the console.
```

* Greatly improve overall phrasing
* Be much more specific about the commands to use and why you would use each backend
* Break up this run-on sentence into more digestible parts
* Avoid much repetition/redundancy and convoluted synatax